dev/tests/integration/testsuite/Magento/LoginAsCustomer/_files/admin.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+use Magento\Authorization\Model\RoleFactory;
+use Magento\Authorization\Model\Role;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\User\Model\User;
+use Magento\Authorization\Model\RulesFactory;
+use Magento\Authorization\Model\Rules;
+
+//Creating a new admin user with a custom role to safely change role settings without affecting the main user's role.
+/** @var Role $role */
+$role = Bootstrap::getObjectManager()->get(RoleFactory::class)->create();
+$role->setName('test_custom_role');
+$role->setData('role_name', $role->getName());
+$role->setRoleType(\Magento\Authorization\Model\Acl\Role\Group::ROLE_TYPE);
+$role->setUserType((string)\Magento\Authorization\Model\UserContextInterface::USER_TYPE_ADMIN);
+$role->save();
+/** @var Rules $rules */
+$rules = Bootstrap::getObjectManager()->get(RulesFactory::class)->create();
+$rules->setRoleId($role->getId());
+//Granted all permissions.
+$rules->setResources([Bootstrap::getObjectManager()->get(\Magento\Framework\Acl\RootResource::class)->getId()]);
+$rules->saveRel();
+
+/** @var User $user */
+$user = Bootstrap::getObjectManager()->create(User::class);
+$user->setFirstname("John")
+    ->setLastname("Doe")
+    ->setUsername('TestAdmin1')
+    ->setPassword(\Magento\TestFramework\Bootstrap::ADMIN_PASSWORD)
+    ->setEmail('testadmin1@gmail.com')
+    ->setIsActive(1)
+    ->setRoleId($role->getId());
+$user->save();
```

Comment:
Please, do not use deprecated methods in your code. (save, delete of ORM models). Use repositories or resource models instead.

Reply:
@rogyar This is used to create user and save that user. We did follow Model approach as I can't see any repository or interface for that. Resource model doesn't have these methods. Additionally I was checking for reference and I am following to https://github.com/magento/magento2/blob/2.4-develop/dev/tests/integration/testsuite/Magento/User/_files/user_with_custom_role.php where Model is used to save.

Please suggest me if you have some better reference for this.

Reply:
You may find an example of using repo for saving customer in the new fixtures. The old fixtures often contain deprecated methods 

With proper IDE configuration (and DocBlocks with type hinting, you should be notified about that)

![image](https://user-images.githubusercontent.com/2148959/99963361-f6429a80-2d91-11eb-93a2-0c96cbcbf7c6.png)

In any case, you may find an example of saving customer via repo here

https://github.com/magento/magento2/blob/a5a98dfa56cac8cdb60e15568c10578f522be3f3/dev/tests/integration/testsuite/Magento/Customer/_files/customer_for_second_store.php#L21

Reply:
@rogyar this is for admin user not for Customer. Is there ant repository available for Admin User? Can I get a reference please for any CRUD operation other than using Magento\User\Model\User.

Reply:
@rogyar updated the PR with suggested changes. Please review, Thank you